api/configuration/v1/kongconsumer_types.go

```diff
@@ -34,6 +34,8 @@ import (
 // +kubebuilder:printcolumn:name="Programmed",type=string,JSONPath=`.status.conditions[?(@.type=="Programmed")].status`
 // +kubebuilder:validation:XValidation:rule="has(self.username) || has(self.custom_id)", message="Need to provide either username or custom_id"
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.controlPlaneRef) || has(self.spec.controlPlaneRef)", message="controlPlaneRef is required once set"
+//  REVIEW: allow same namespace?
```

Comment:
Yes please, if possible let's add that rule.

Reply:
Looks like we cannot access `metadata.namespace` in CEL: https://github.com/kubernetes/kubernetes/issues/122163.

Reply:
So from https://github.com/kubernetes/kubernetes/issues/122163#issuecomment-1908351389 it seems that it is doable with a different order of fields in the generated CRD manifest but apparently kubebuilder doesn't generate the CRDs in this way?

Nonetheless, we don't actually need to allow specifying the namespace here. Users would simply not provide one when the referenced entity exists in the same namespace as the referrer.